openquake/commonlib/readinput.py

```diff
@@ -420,15 +421,19 @@ def get_site_collection(oqparam):
         except ValueError:
             # this is the normal case
             depth = None
-        sitecol = site.SiteCollection.from_points(
-            sm['lon'], sm['lat'], depth, sm, req_site_params)
-        if oqparam.region_grid_spacing:
-            logging.info('Reducing the grid sites to the site '
-                         'parameters within the grid spacing')
+        if grid_spacing:
+            logging.warning('Having a grid and a site model at the same time '
```

Comment:
There's already a similar warning for the case where a grid, a site model, and an exposure model are provided: https://github.com/gem/oq-engine/blob/engine-3.5/openquake/commonlib/oqvalidation.py#L307-L312

Can the two warnings be combined?

Another note: `oq prepare_site_model` works with exposure and sites files, but not yet with a region + region grid spacing. Won't this advice be confusing to the user?

Reply:
Damn, I forgot that I had already the warning in place, so I will only keep the original warning.

`oq prepare_site_model` works with a `--grid-spacing` option: if you will pass it, it will grid the convex hull of the exposure.

Reply:
Yes, though this case is different, in that there is no exposure. The user has only a `region` (defined by WKT polygon vertices) and a `region_grid_spacing` for which they wish to compute ground motion fields. In addition they have site parameters on a separate regular grid / irregular set of points.

Reply:
This case should work already. Can you add a test in oq-risk-tests?